rust-runtime/aws-smithy-runtime/src/client/http/body/minimum_throughput/throughput.rs

```diff
@@ -343,8 +347,24 @@ impl ThroughputLogs {
         assert!(self.current_tail >= now);
     }
 
+    /// Mark the stream complete indicating no more data is expected. This is an
+    /// idempotent operation -- subsequent invocations of this function have no effect
+    /// and return false.
+    ///
+    /// After marking a stream complete [report](#method.report) will forever more return
+    /// [ThroughputReport::Complete]
+    pub(super) fn complete(&mut self) -> bool {
```

Comment:
Could we name this `mark_complete` or something to indicate that it has a side effect? When I saw this method used in `UploadThroughput` I assumed it was just checking the completeness, not setting it. And with that the `complete` method in `UploadThroughput` should probably be renamed as well.